Runtime/Controller/JointControl.cs

```diff
@@ -39,12 +39,62 @@ void Start()
             {
                 ArticulationDrive currentDrive = joint.xDrive;
                 float newTargetDelta = (int)direction * Time.fixedDeltaTime * speed;
-                if (newTargetDelta + currentDrive.target <= currentDrive.upperLimit && newTargetDelta + currentDrive.target >= currentDrive.lowerLimit){
+                if (((joint.twistLock == ArticulationDofLock.LimitedMotion) && (newTargetDelta + currentDrive.target <= currentDrive.upperLimit && newTargetDelta + currentDrive.target >= currentDrive.lowerLimit)) || 
+                (joint.twistLock == ArticulationDofLock.FreeMotion)){
                     currentDrive.target += newTargetDelta;
-                } 
+                }
+
+                if(joint.jointType == ArticulationJointType.RevoluteJoint)
+                {
+                    if(joint.twistLock == ArticulationDofLock.LimitedMotion)
+                    {
+                        if(newTargetDelta + currentDrive.target > currentDrive.upperLimit)
+                        {
+                            currentDrive.target = currentDrive.upperLimit;
+                        }
+                        else if(newTargetDelta + currentDrive.target < currentDrive.lowerLimit)
+                        {
+                            currentDrive.target = currentDrive.lowerLimit;
+                        }
+                        else
+                        {
+                            currentDrive.target += newTargetDelta;
+                        }
+                    }
+                    else
+                    {
+                        currentDrive.target += newTargetDelta;
+   
+                    }
+                }
+
+                else if(joint.jointType == ArticulationJointType.PrismaticJoint)
```

Comment:
shall we combine this line and line 47 to `if (joint.jointType == ArticulationJointType.RevoluteJoint || joint.jointType == ArticulationJointType.PrismaticJoint)` ?

Reply:
they both have different conditions in them. For prismatic joint, it is linear limit lock and for revolute joint its twist lock.